ccl/changefeedccl: TestChangefeedSchemaTTL failed

ccl/changefeedccl.TestChangefeedSchemaTTL [failed](https://mesolite.cluster.engflow.com/invocations/default/3f4ede90-eb1c-44ac-9dbd-f8705319812d?testReportRun=2&testReportShard=4&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [20aa5054030c4e802024e734ea588db21257ab3a](https://github.com/cockroachdb/cockroach/commits/20aa5054030c4e802024e734ea588db21257ab3a):


```
=== RUN   TestChangefeedSchemaTTL
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestChangefeedSchemaTTL3625926609
    test_log_scope.go:76: use -show-logs to present logs inline
    changefeed_test.go:6544: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedSchemaTTL3625926609
--- FAIL: TestChangefeedSchemaTTL (36.93s)
=== RUN   TestChangefeedSchemaTTL/sinkless
    helpers_test.go:1241: making server as system tenant
    helpers_test.go:1317: making sinkless feed factory
    helpers_test.go:1382: pgURL sinkless SinklessFeedUser
    helpers_test.go:1382: pgURL sinkless root
    testfeed_test.go:294: sinkless feed creating changefeed: CREATE CHANGEFEED FOR TABLE foo
    testfeed_test.go:245: sinkless feed closing connection due to timeout (30s): CREATE CHANGEFEED FOR TABLE foo
    changefeed_test.go:6534: 
        	Error Trace:	pkg/ccl/changefeedccl/changefeed_test.go:6534
        	            				pkg/ccl/changefeedccl/helpers_test.go:1524
        	Error:      	Expect "read tcp 127.0.0.1:39922->127.0.0.1:34233: use of closed network connection" to match "GC threshold"
        	Test:       	TestChangefeedSchemaTTL/sinkless
    testfeed_test.go:301: closing sinkless feed
--- FAIL: TestChangefeedSchemaTTL/sinkless (36.92s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=4</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #148907 ccl/changefeedccl: TestChangefeedSchemaTTL failed [A-cdc C-test-failure O-robot T-cdc branch-release-25.3 release-blocker]
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedSchemaTTL.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-52029

ccl/changefeedccl.TestChangefeedSchemaTTL [failed](https://tanzanite.cluster.engflow.com/invocations/default/f58b2e1e-8c66-46f5-b1e9-0e92b1c4a0fc?testReportRun=2&testReportShard=4&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [e5d1c9413dc322f0cf1cfc4da99e9504a1e9ddc1](https://github.com/cockroachdb/cockroach/commits/e5d1c9413dc322f0cf1cfc4da99e9504a1e9ddc1):


```
=== RUN   TestChangefeedSchemaTTL
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestChangefeedSchemaTTL1889218443
    test_log_scope.go:76: use -show-logs to present logs inline
    changefeed_test.go:6544: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedSchemaTTL1889218443
--- FAIL: TestChangefeedSchemaTTL (37.66s)
=== RUN   TestChangefeedSchemaTTL/sinkless
    helpers_test.go:1241: making server as system tenant
    helpers_test.go:1317: making sinkless feed factory
    helpers_test.go:1382: pgURL sinkless SinklessFeedUser
    helpers_test.go:1382: pgURL sinkless root
    helpers_test.go:1053: did not force enriched envelope for CREATE CHANGEFEED FOR TABLE foo because *changefeedccl.sinklessFeedFactory is not supported
    testfeed_test.go:294: sinkless feed creating changefeed: CREATE CHANGEFEED FOR TABLE foo
    testfeed_test.go:245: sinkless feed closing connection due to timeout (30s): CREATE CHANGEFEED FOR TABLE foo
    changefeed_test.go:6534: 
        	Error Trace:	pkg/ccl/changefeedccl/changefeed_test.go:6534
        	            				pkg/ccl/changefeedccl/helpers_test.go:1524
        	Error:      	Expect "read tcp 127.0.0.1:45412->127.0.0.1:33107: use of closed network connection" to match "GC threshold"
        	Test:       	TestChangefeedSchemaTTL/sinkless
    testfeed_test.go:301: closing sinkless feed
--- FAIL: TestChangefeedSchemaTTL/sinkless (37.65s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=4</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #148907 ccl/changefeedccl: TestChangefeedSchemaTTL failed [A-cdc C-test-failure O-robot T-cdc branch-release-25.3 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedSchemaTTL.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedSchemaTTL [failed](https://mesolite.cluster.engflow.com/invocations/default/51792da6-71e8-4838-8f7d-4c0e7dc91959?testReportRun=3&testReportShard=7&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [afef2d239966541de7611883fbb803f08c0fed92](https://github.com/cockroachdb/cockroach/commits/afef2d239966541de7611883fbb803f08c0fed92):


```
=== RUN   TestChangefeedSchemaTTL
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestChangefeedSchemaTTL2005575833
    test_log_scope.go:82: use -show-logs to present logs inline
    changefeed_test.go:6729: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedSchemaTTL2005575833
--- FAIL: TestChangefeedSchemaTTL (36.75s)
=== RUN   TestChangefeedSchemaTTL/sinkless
    helpers_test.go:1297: making server as system tenant
    helpers_test.go:1373: making sinkless feed factory
    helpers_test.go:1438: pgURL sinkless SinklessFeedUser
    helpers_test.go:1438: pgURL sinkless root
    testfeed_test.go:294: sinkless feed creating changefeed: CREATE CHANGEFEED FOR TABLE foo
    testfeed_test.go:245: sinkless feed closing connection due to timeout (30s): CREATE CHANGEFEED FOR TABLE foo
    changefeed_test.go:6719: 
        	Error Trace:	pkg/ccl/changefeedccl/changefeed_test.go:6719
        	            				pkg/ccl/changefeedccl/helpers_test.go:1580
        	Error:      	Expect "read tcp 127.0.0.1:58000->127.0.0.1:38513: use of closed network connection" to match "GC threshold"
        	Test:       	TestChangefeedSchemaTTL/sinkless
    testfeed_test.go:301: closing sinkless feed
--- FAIL: TestChangefeedSchemaTTL/sinkless (36.75s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=7</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #150549 ccl/changefeedccl: TestChangefeedSchemaTTL failed [A-cdc C-test-failure O-robot T-cdc X-duplicate branch-release-25.3.0-rc]
- #148907 ccl/changefeedccl: TestChangefeedSchemaTTL failed [A-cdc C-test-failure O-robot P-2 T-cdc branch-release-25.3]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedSchemaTTL.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedSchemaTTL [failed](https://mesolite.cluster.engflow.com/invocations/default/0f09732a-a4e5-4014-8409-8bae5e64cd59?testReportRun=1&testReportShard=7&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [9831be5f895ec7758d43bbec0df76f1e14e78d44](https://github.com/cockroachdb/cockroach/commits/9831be5f895ec7758d43bbec0df76f1e14e78d44):


```
=== RUN   TestChangefeedSchemaTTL
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestChangefeedSchemaTTL4101481035
    test_log_scope.go:82: use -show-logs to present logs inline
    changefeed_test.go:6731: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedSchemaTTL4101481035
--- FAIL: TestChangefeedSchemaTTL (37.09s)
=== RUN   TestChangefeedSchemaTTL/sinkless
    helpers_test.go:1386: making server as system tenant
    helpers_test.go:1462: making sinkless feed factory
    helpers_test.go:1535: pgURL sinkless SinklessFeedUser
    helpers_test.go:1535: pgURL sinkless root
    testfeed_test.go:294: sinkless feed creating changefeed: CREATE CHANGEFEED FOR TABLE foo
    testfeed_test.go:245: sinkless feed closing connection due to timeout (30s): CREATE CHANGEFEED FOR TABLE foo
    changefeed_test.go:6721: 
        	Error Trace:	pkg/ccl/changefeedccl/changefeed_test.go:6721
        	            				pkg/ccl/changefeedccl/helpers_test.go:1678
        	Error:      	Expect "read tcp 127.0.0.1:47616->127.0.0.1:36619: use of closed network connection" to match "GC threshold"
        	Test:       	TestChangefeedSchemaTTL/sinkless
    testfeed_test.go:301: closing sinkless feed
--- FAIL: TestChangefeedSchemaTTL/sinkless (37.09s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=7</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #150549 ccl/changefeedccl: TestChangefeedSchemaTTL failed [A-cdc C-test-failure O-robot T-cdc X-duplicate branch-release-25.3.0-rc]
- #148907 ccl/changefeedccl: TestChangefeedSchemaTTL failed [A-cdc C-test-failure O-robot P-2 T-cdc branch-release-25.3]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedSchemaTTL.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedSchemaTTL [failed](https://mesolite.cluster.engflow.com/invocations/default/9fbb1ab1-87a0-4163-85b6-0a1bd4315368?testReportRun=3&testReportShard=7&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [a4ad94ae1379daa20b467c89d09de27ba8e2356d](https://github.com/cockroachdb/cockroach/commits/a4ad94ae1379daa20b467c89d09de27ba8e2356d):


```
=== RUN   TestChangefeedSchemaTTL
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestChangefeedSchemaTTL3751225986
    test_log_scope.go:82: use -show-logs to present logs inline
    changefeed_test.go:6731: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedSchemaTTL3751225986
--- FAIL: TestChangefeedSchemaTTL (36.66s)
=== RUN   TestChangefeedSchemaTTL/sinkless
    helpers_test.go:1386: making server as system tenant
    helpers_test.go:1462: making sinkless feed factory
    helpers_test.go:1535: pgURL sinkless SinklessFeedUser
    helpers_test.go:1535: pgURL sinkless root
    testfeed_test.go:294: sinkless feed creating changefeed: CREATE CHANGEFEED FOR TABLE foo
    testfeed_test.go:245: sinkless feed closing connection due to timeout (30s): CREATE CHANGEFEED FOR TABLE foo
    changefeed_test.go:6721: 
        	Error Trace:	pkg/ccl/changefeedccl/changefeed_test.go:6721
        	            				pkg/ccl/changefeedccl/helpers_test.go:1678
        	Error:      	Expect "read tcp 127.0.0.1:58066->127.0.0.1:41573: use of closed network connection" to match "GC threshold"
        	Test:       	TestChangefeedSchemaTTL/sinkless
    testfeed_test.go:301: closing sinkless feed
--- FAIL: TestChangefeedSchemaTTL/sinkless (36.66s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=7</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #150549 ccl/changefeedccl: TestChangefeedSchemaTTL failed [A-cdc C-test-failure O-robot T-cdc X-duplicate branch-release-25.3.0-rc]
- #148907 ccl/changefeedccl: TestChangefeedSchemaTTL failed [A-cdc C-test-failure O-robot P-2 T-cdc branch-release-25.3]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedSchemaTTL.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedSchemaTTL [failed](https://mesolite.cluster.engflow.com/invocations/default/46b2f637-a508-44d2-8b29-9b2587231602?testReportRun=2&testReportShard=11&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [e3770b8d2034d7eeaa5095a3b2214da3f35b895e](https://github.com/cockroachdb/cockroach/commits/e3770b8d2034d7eeaa5095a3b2214da3f35b895e):


```
=== RUN   TestChangefeedSchemaTTL
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestChangefeedSchemaTTL3662330067
    test_log_scope.go:82: use -show-logs to present logs inline
    changefeed_test.go:6884: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedSchemaTTL3662330067
--- FAIL: TestChangefeedSchemaTTL (35.99s)
=== RUN   TestChangefeedSchemaTTL/sinkless
    helpers_test.go:1393: making server as system tenant
    helpers_test.go:1469: making sinkless feed factory
    helpers_test.go:1542: pgURL sinkless SinklessFeedUser
    helpers_test.go:1542: pgURL sinkless root
    testfeed_test.go:311: sinkless feed creating changefeed: CREATE CHANGEFEED FOR TABLE foo
    testfeed_test.go:262: sinkless feed closing connection due to timeout (30s): CREATE CHANGEFEED FOR TABLE foo
    changefeed_test.go:6874: 
        	Error Trace:	pkg/ccl/changefeedccl/changefeed_test.go:6874
        	            				pkg/ccl/changefeedccl/helpers_test.go:1685
        	Error:      	Expect "read tcp 127.0.0.1:53422->127.0.0.1:42625: use of closed network connection" to match "GC threshold"
        	Test:       	TestChangefeedSchemaTTL/sinkless
    testfeed_test.go:318: closing sinkless feed
--- FAIL: TestChangefeedSchemaTTL/sinkless (35.98s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=11</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #152656 ccl/changefeedccl: TestChangefeedSchemaTTL failed [A-cdc C-test-failure O-robot P-2 T-cdc X-duplicate branch-release-25.3.1-rc]
- #150549 ccl/changefeedccl: TestChangefeedSchemaTTL failed [A-cdc C-test-failure O-robot P-2 T-cdc X-duplicate branch-release-25.3.0-rc]
- #148907 ccl/changefeedccl: TestChangefeedSchemaTTL failed [A-cdc C-test-failure O-robot P-2 T-cdc branch-release-25.3]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedSchemaTTL.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedSchemaTTL [failed](https://mesolite.cluster.engflow.com/invocations/default/b2498c71-fd95-4d91-9d4c-0655c6d6bb2a?testReportRun=3&testReportShard=11&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [b8197d505b173f1e44e68f72a8bb679959923210](https://github.com/cockroachdb/cockroach/commits/b8197d505b173f1e44e68f72a8bb679959923210):


```
=== RUN   TestChangefeedSchemaTTL
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestChangefeedSchemaTTL3324183744
    test_log_scope.go:82: use -show-logs to present logs inline
    changefeed_test.go:6884: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedSchemaTTL3324183744
--- FAIL: TestChangefeedSchemaTTL (36.93s)
=== RUN   TestChangefeedSchemaTTL/sinkless
    helpers_test.go:1394: making server as system tenant
    helpers_test.go:1470: making sinkless feed factory
    helpers_test.go:1543: pgURL sinkless SinklessFeedUser
    helpers_test.go:1543: pgURL sinkless root
    testfeed_test.go:311: sinkless feed creating changefeed: CREATE CHANGEFEED FOR TABLE foo
    testfeed_test.go:262: sinkless feed closing connection due to timeout (30s): CREATE CHANGEFEED FOR TABLE foo
    changefeed_test.go:6874: 
        	Error Trace:	pkg/ccl/changefeedccl/changefeed_test.go:6874
        	            				pkg/ccl/changefeedccl/helpers_test.go:1686
        	Error:      	Expect "read tcp 127.0.0.1:34926->127.0.0.1:41169: use of closed network connection" to match "GC threshold"
        	Test:       	TestChangefeedSchemaTTL/sinkless
    testfeed_test.go:318: closing sinkless feed
--- FAIL: TestChangefeedSchemaTTL/sinkless (36.93s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=11</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #152656 ccl/changefeedccl: TestChangefeedSchemaTTL failed [A-cdc C-test-failure O-robot P-2 T-cdc X-duplicate branch-release-25.3.1-rc]
- #150549 ccl/changefeedccl: TestChangefeedSchemaTTL failed [A-cdc C-test-failure O-robot P-2 T-cdc X-duplicate branch-release-25.3.0-rc]
- #148907 ccl/changefeedccl: TestChangefeedSchemaTTL failed [A-cdc C-test-failure O-robot P-2 T-cdc branch-release-25.3]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedSchemaTTL.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedSchemaTTL [failed](https://mesolite.cluster.engflow.com/invocations/default/52f036a2-3b0c-45ca-b450-b8640faba6d2?testReportRun=3&testReportShard=11&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [8b1e1a9e399ca84a87c686d0106279e0c5904871](https://github.com/cockroachdb/cockroach/commits/8b1e1a9e399ca84a87c686d0106279e0c5904871):


```
=== RUN   TestChangefeedSchemaTTL
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestChangefeedSchemaTTL3640030915
    test_log_scope.go:82: use -show-logs to present logs inline
    changefeed_test.go:6884: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedSchemaTTL3640030915
--- FAIL: TestChangefeedSchemaTTL (37.20s)
=== RUN   TestChangefeedSchemaTTL/sinkless
    helpers_test.go:1394: making server as system tenant
    helpers_test.go:1470: making sinkless feed factory
    helpers_test.go:1543: pgURL sinkless SinklessFeedUser
    helpers_test.go:1543: pgURL sinkless root
    helpers_test.go:1198: did not force enriched envelope for CREATE CHANGEFEED FOR TABLE foo because *changefeedccl.sinklessFeedFactory is not supported
    testfeed_test.go:311: sinkless feed creating changefeed: CREATE CHANGEFEED FOR TABLE foo
    testfeed_test.go:262: sinkless feed closing connection due to timeout (30s): CREATE CHANGEFEED FOR TABLE foo
    changefeed_test.go:6874: 
        	Error Trace:	pkg/ccl/changefeedccl/changefeed_test.go:6874
        	            				pkg/ccl/changefeedccl/helpers_test.go:1686
        	Error:      	Expect "read tcp 127.0.0.1:52902->127.0.0.1:40473: use of closed network connection" to match "GC threshold"
        	Test:       	TestChangefeedSchemaTTL/sinkless
    testfeed_test.go:318: closing sinkless feed
--- FAIL: TestChangefeedSchemaTTL/sinkless (37.19s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=11</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #153138 ccl/changefeedccl: TestChangefeedSchemaTTL failed [A-cdc C-test-failure O-robot T-cdc branch-release-25.3.2-rc release-blocker]
- #152656 ccl/changefeedccl: TestChangefeedSchemaTTL failed [A-cdc C-test-failure O-robot P-2 T-cdc X-duplicate branch-release-25.3.1-rc]
- #150549 ccl/changefeedccl: TestChangefeedSchemaTTL failed [A-cdc C-test-failure O-robot P-2 T-cdc X-duplicate branch-release-25.3.0-rc]
- #148907 ccl/changefeedccl: TestChangefeedSchemaTTL failed [A-cdc C-test-failure O-robot P-2 T-cdc branch-release-25.3]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedSchemaTTL.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedSchemaTTL [failed](https://mesolite.cluster.engflow.com/invocations/default/da187a2c-8a9b-4a25-a4a7-223c01b4b112?testReportRun=2&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [2e14011ff0af49f901a8347f85c4246bd179d119](https://github.com/cockroachdb/cockroach/commits/2e14011ff0af49f901a8347f85c4246bd179d119):


```
=== RUN   TestChangefeedSchemaTTL
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestChangefeedSchemaTTL2109464701
    test_log_scope.go:82: use -show-logs to present logs inline
    changefeed_test.go:7097: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedSchemaTTL2109464701
--- FAIL: TestChangefeedSchemaTTL (37.30s)
=== RUN   TestChangefeedSchemaTTL/sinkless
    helpers_test.go:1394: making server as system tenant
    helpers_test.go:1470: making sinkless feed factory
    helpers_test.go:1543: pgURL sinkless SinklessFeedUser
    helpers_test.go:1543: pgURL sinkless root
    helpers_test.go:1198: did not force enriched envelope for CREATE CHANGEFEED FOR TABLE foo because *changefeedccl.sinklessFeedFactory is not supported
    testfeed_test.go:311: sinkless feed creating changefeed: CREATE CHANGEFEED FOR TABLE foo
    testfeed_test.go:262: sinkless feed closing connection due to timeout (30s): CREATE CHANGEFEED FOR TABLE foo
    changefeed_test.go:7087: 
        	Error Trace:	pkg/ccl/changefeedccl/changefeed_test.go:7087
        	            				pkg/ccl/changefeedccl/helpers_test.go:1692
        	Error:      	Expect "read tcp 127.0.0.1:32818->127.0.0.1:39073: use of closed network connection" to match "GC threshold"
        	Test:       	TestChangefeedSchemaTTL/sinkless
    testfeed_test.go:318: closing sinkless feed
--- FAIL: TestChangefeedSchemaTTL/sinkless (37.29s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #154304 ccl/changefeedccl: TestChangefeedSchemaTTL failed [A-cdc C-test-failure O-robot P-1 T-cdc branch-release-25.3 release-blocker]
- #154229 ccl/changefeedccl: TestChangefeedSchemaTTL failed [A-cdc C-test-failure O-robot P-1 T-cdc branch-release-25.4]
- #153138 ccl/changefeedccl: TestChangefeedSchemaTTL failed [A-cdc C-test-failure O-robot P-2 T-cdc X-duplicate branch-release-25.3.2-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedSchemaTTL.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/ccl/changefeedccl/changefeed_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_test.go
@@ -7043,6 +7043,7 @@ func TestChangefeedSchemaTTL(t *testing.T) {
 		// Create the data table; it will only contain a single row with multiple
 		// versions.
 		sqlDB.Exec(t, `CREATE TABLE foo (a INT PRIMARY KEY, b STRING)`)
+		sqlDB.Exec(t, `CREATE USER joe`)
 
 		counter := 0
 		upsertRow := func() {
@@ -7072,6 +7073,15 @@ func TestChangefeedSchemaTTL(t *testing.T) {
 		// Force a GC of the table. This should cause both older versions of the
 		// table to be deleted, with the middle version being lost to the changefeed.
 		forceTableGC(t, s.SystemServer, sqlDB, "system", "descriptor")
+
+		// Do an unnecessary version bump on the descriptor, which will purge old
+		// versions of the descriptor that the lease manager may have cached for
+		// historical queries. When schema_locked support is active, the act of
+		// detecting polling inside pauseOrResumePolling can cause old versions
+		// to be cached. The historical descriptors are cleared everytime a
+		// version bump occurs, otherwise this test can flake if the prior versions
+		// need to decode a row is already cached.
+		waitForSchemaChange(t, sqlDB, "ALTER TABLE foo OWNER TO joe")
 
 		// Resume our changefeed normally.
 		atomic.StoreInt32(&shouldWait, 0)

